src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -279,6 +285,36 @@ public function addClassResource(\ReflectionClass $class)
     }
 
     /**
+     * Retrieves the requested reflection class and registers it for resource tracking.
+     *
+     * @param string $class
+     *
+     * @return \ReflectionClass|false
```

Comment:
@nicolas-grekas I think we need to try to not use mixed return args in our public APIs (anymore). With `\ReflectionClass|null`, we could add a native PHP return type in 7.1, but `\ReflectionClass|false`, we cannot.